Low cloud tracking tobathon

JuliaKukulies opened the pull request:
Our work from Tobathon2025, solves issue #483 

This PR adds a new example jupyter notebook that demonstrate how tobac can be used to track shallow cumulus clouds using LES data from the [Cloud Botany]( https://howto.eurec4a.eu/botany_dales.html) project. 

This notebook follows the typical tracking steps feature detection, tracking and segmentation and includes an exmaple for a simple lifecycle analysis with the output data. 

Thanks to Will Jones (@w-k-jones) and Milind Sharma

JuliaKukulies commented:
@w-k-jones I have cleaned up the notebook and added some text. Please have a look and modify further if you want to change anything before we merge this. And do you know Milind's github username?

w-k-jones commented:
> @w-k-jones I have cleaned up the notebook and added some text. Please have a look and modify further if you want to change anything before we merge this. And do you know Milind's github username?

Great, thanks, I really like the visualisations you've added!

And Milind's github is @gewitterblitz

JuliaKukulies commented:
> @JuliaKukulies I agree, the cleaned-up notebook looks ready to be added as an example for tracking shallow clouds.
> 
> Thank you so much to you and @w-k-jones for leading this effort!

Great, thanks for your feedback @gewitterblitz !

JuliaKukulies commented:
> > @w-k-jones I have cleaned up the notebook and added some text. Please have a look and modify further if you want to change anything before we merge this. And do you know Milind's github username?
> 
> Great, thanks, I really like the visualisations you've added! I'll have a look through any final changes on Monday
> 
> And Milind's github is @gewitterblitz

Sounds good @w-k-jones and thanks for your feedback! I will also need to check why the notebook checks fail here before we merge this

JuliaKukulies commented:
@w-k-jones do you have any more changes that you wanted to add to this notebook? 

Also, the checks fail because of the missing library `intake`. Would you suggest to just add the additional dependencies in `example_requirements.txt` or `environment-ci.yml` or both?

w-k-jones commented:
> @w-k-jones do you have any more changes that you wanted to add to this notebook?
> 
> Also, the checks fail because of the missing library `intake` (and `eurec4a`). Would you suggest to just add the additional dependencies in `example_requirements.txt` or `environment-ci.yml` or both?

Yep, I think both `intake==0.7.0` and `intake-xarray==0.7.0` are required, I'll add them to both and check that it works

w-k-jones commented:
Right, sorry for the delays in getting round to sorting the final changes here, I'm going to aim to have this notebook included in the v1.6 release

freemansw1 commented:
@w-k-jones @JuliaKukulies Thanks for your work on this. Feel free to ping when you're ready for review.

JuliaKukulies commented:
> @w-k-jones @JuliaKukulies Thanks for your work on this. Feel free to ping when you're ready for review.

Good for review from my side, when @w-k-jones pushes his changes and updated package requirements to this branch

w-k-jones commented:
One other question, do we want to merge this into `tobathon_2025` or directly into `main`?

w-k-jones commented:
@JuliaKukulies I've updated the example requirements, and made sure that the notebook runs correctly. I'm happy for this to be made ready for review if you are :)

w-k-jones commented:
I've changed the merge target to `RC_v1.6.x`, hopefully we get this out in one of the upcoming releases

JuliaKukulies commented:
> I've changed the merge target to `RC_v1.6.x`, hopefully we get this out in one of the upcoming releases

That makes sense, thanks, Will! I have fixed the formatting issues that previously let the tests fail and I merged in the latest changes from RC_v.1.6.x which I believe were updated after the last release. Still there seems to be some conflicts with the package setups that make the whole CI fail. Do you know how we can fix this? 

After fixing the tests, I am happy to open this for review and get it merged very soon.

JuliaKukulies commented:
> Tracked down the source of the workflow issue: `requirements.txt` had been reformatted at some point which broke the install. Not sure how this happened, but might be worth checking what black is doing in future and whether it now applies beyond .py files. @fsenf @kelcyno now ready for review :)

Oh wow, thanks for finding out, Will! I must have caused this by manually applying black after the first formatting was not sufficient, not sure. But I am glad you found the issue and I will check if I can reproduce this by running black manually on all files and if there is something we could do to prevent this.
